### PR TITLE
IN clause support in hql for composite keys on databases without row value constructor support

### PR DESCRIPTION
**Pull request by bahusoid**
Port hibernate [`mutateRowValueConstructorSyntaxInInListSyntax`](https://github.com/hibernate/hibernate-orm/blob/e5dc635a52362f69b69acb8d5b166b69b165dbbd/hibernate-core/src/main/java/org/hibernate/hql/internal/ast/tree/InLogicOperatorNode.java#L119).

Allows to use `IN (compositeId1, compositeId2)`  in hql  (and `Contains` queries in LINQ) for DBs like SQL Server.

~Depends on #2211 (without it "mutate" logic applies for all dialects)~

### CODE REVIEW

**Comment by hazzik** [comment]
>Depends on #2158 ( SupportsRowValueConstructorSyntaxInInList part)

Can you please unwind this dependency?

File: src/NHibernate/Hql/Ast/ANTLR/Tree/BinaryLogicOperatorNode.cs [hunk 1]
```diff
@@ -200,17 +200,19 @@ public IParameterSpecification[] GetEmbeddedParameters()
 			return embeddedParameters.ToArray();
 		}
 
-		private string Translate(int valueElements, string comparisonText, string[] lhsElementTexts, string[] rhsElementTexts)
+		//TODO 6.0: Change to private protected (C#7.2 feature)
```
Comment (hazzik) [inline src/NHibernate/Hql/Ast/ANTLR/Tree/BinaryLogicOperatorNode.cs]: You can change the project settings. To enable 7.2. I think the build agent should handle it.

**Comment by bahusoid** [comment]
> Can you please unwind this dependency?
I created separate PR #2211

File: src/NHibernate/Hql/Ast/ANTLR/Tree/BinaryLogicOperatorNode.cs (hunk 1 above)
Comment (bahusoid) [reply src/NHibernate/Hql/Ast/ANTLR/Tree/BinaryLogicOperatorNode.cs]: Done. Will see if all is OK with CI builds

**Pull request merged**